source/grid/manifold.cc

```diff
@@ -512,6 +512,15 @@ FlatManifold<dim,spacedim>::FlatManifold (const Tensor<1,spacedim> &periodicity,
 
 
 
+template<int dim, int spacedim>
+std::unique_ptr<Manifold<dim, spacedim> >
+FlatManifold<dim,spacedim>::clone() const
+{
+  return std::unique_ptr<Manifold<dim,spacedim> >(new FlatManifold<dim,spacedim>(periodicity, tolerance));
```

Comment:
`std_cxx14::make_unique`. Likely also further down in the patch in a number of places, but I won't mark these up individually.

Reply:
Or are you doing this because of the mismatch of the static and dynamic type of the object pointed to?

Reply:
make_unique would work also in that case, as unique_ptrs know how to convert from "pointers to derived" to "pointers to base", but I preferred to have it this way as this documents better what is happening. I know you are fond of `make_unique`. I'll change these... ;)

Reply:
Appreciated :-) Though it's not actually that I'm particularly fond of `make_unique`. It's more that we seem to be moving away from explicitly calling `operator new` recently. I don't remember who started this, I've just hopped on the train. I do think that not using `new` is useful in identifying the places that we haven't yet converted to using `std::unique/shared_ptr` and that could profit from this.